types/project.go

```diff
@@ -321,6 +350,35 @@ func (p *Project) ForServices(names []string) error {
 	return nil
 }
 
+// ForServicesAndDependent restrict the project model to selected services and those who depend on them
+func (p *Project) ForServicesAndDependent(names []string) error {
```

Comment:
I wonder if we could do this with options to `ForServices`, e.g.
```go
func (p *Project) ForServices(names []string, opt... ServiceFilterOption) error {
    ...
}

p.ForServices([]string{"foo", "bar"}, IncludeDependents)
```
(Could do the same for `WithServices` as well.)

Otherwise, small nit here: should be `ForServicesAndDependents` (plural)